tests/libclient/libclient.go

```diff
@@ -18,3 +18,29 @@
  */
 
 package libclient
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+const waitingVMInstanceStart = "Waiting until the VirtualMachineInstance will start"
+
+func RunVMIAndExpectLaunch(vmi *v1.VirtualMachineInstance, timeout int) *v1.VirtualMachineInstance {
+	vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
I believe its better to make this function return the error and do the expect in from where its been called.

Reply:
Although I see that the WaitForVMIPhase has their own Expect.. So maybe at least use:
g.ExpectWithOffset(1, err).ToNot(gomega.HaveOccurred())

Reply:
I like your idea, but I prefer we do this on a separate PR. 
This PR only moves a function - syntactic change, no logic.

Reply:
IMHO it is better to fix things while handling them.
An improvement in providing errors does not affect any readability or commit scope.
I vote for `ExpectWithOffset`.

Reply:
I'll add this change in a separate commit then.

Reply:
yeah I mentioned that WaitForVMIPhase works similarly having expect inside. So in order to still get relevant information from the calling function we can only replace: 
Expect(err).ToNot(HaveOccurred())
With:
ExpectWithOffset(1, err).ToNot(gomega.HaveOccurred())
So the function will stay the same just bubble up the error source.

Reply:
Fixed. PTAL